### PR TITLE
[Sparse] Add GCN example

### LINKED ISSUE #4490: Sparse Ops: GCN example

### PR DESCRIPTION
**Pull request by czkkkkkk**
## Description
Resolve #4490 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/mock_sparse/gcn.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import torch.nn.functional as F
+import dgl
+from dgl.data import CoraGraphDataset, CiteseerGraphDataset, PubmedGraphDataset
+from dgl import AddSelfLoop
+import argparse
+from torch.nn import init
+
+from dgl.mock_sparse import create_from_coo, diag, identity
+
+from synthetic import SyntheticDataset
```
Comment (jermainewang) [inline examples/pytorch/mock_sparse/gcn.py:12]: Remove SyntheticDataset and the corresponding file.

**Review by jermainewang — APPROVED** [review]
Approved. But could you add a README about how to run the example and the expected outcome?

File: examples/pytorch/mock_sparse/gcn.py (hunk 1 above)
Comment (czkkkkkk) [reply examples/pytorch/mock_sparse/gcn.py]: Done

**Comment by czkkkkkk** [comment]
> Approved. But could you add a README about how to run the example and the expected outcome?

Added a README following the style of original GCN example.

**Review by jermainewang — APPROVED** [review]

File: examples/pytorch/mock_sparse/gcn/train.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            h = layer(A, h)
+        return h
+
+
+def gcn_norm(A):
+    # normalization
+    I = identity(A.shape)  # create an identity matrix
+    A_hat = A + I  # add self-loop to A
+    D = diag(A_hat.sum(0))  # diagonal degree matrix of A_hat
+    # FIXME DiagMatrix does not have power() method
```
Comment (isratnisa) [inline examples/pytorch/mock_sparse/gcn/train.py:58]: It should have now

**Pull request merged**